Sources/Widgets/Widgets3D/AngleWidget/behavior.js

```diff
@@ -85,20 +87,24 @@ export default function widgetBehavior(publicAPI, model) {
       model.activeState.getActive() &&
       !ignoreKey(callData)
     ) {
-      const { worldCoords } = manipulator.handleEvent(
+      const { worldCoords, worldDelta } = manipulator.handleEvent(
         callData,
         model._apiSpecificRenderWindow
       );
 
-      if (
-        worldCoords.length &&
-        (model.activeState === model.widgetState.getMoveHandle() ||
-          model._isDragging) &&
-        model.activeState.setOrigin // e.g. the line is pickable but not draggable
-      ) {
-        model.activeState.setOrigin(worldCoords);
-        publicAPI.invokeInteractionEvent();
-        return macro.EVENT_ABORT;
+      if (worldCoords.length && model.activeState.setOrigin) {
+        if (model.activeState === model.widgetState.getMoveHandle()) {
+          model.activeState.setOrigin(worldCoords);
+          publicAPI.invokeInteractionEvent();
+          return macro.EVENT_ABORT;
+        }
+        if (model._isDragging) {
+          model.activeState.setOrigin(
+            add(model.activeState.getOrigin(), worldDelta, [])
```

Comment:
please explain in comments why here you use `worldDelta`, but not if `activeState === model.widgetState.getModeHandle)`. It is not obvious to me.

Reply:
The move handle can have a null origin, so `add(moveHandle.getOrigin(), worldDelta, [])` crashes. I'll add a comment about it.